smarts/core/agent_manager.py

```diff
@@ -209,17 +207,13 @@ def _vehicle_reward(self, vehicle_id, sim):
     def _vehicle_score(self, vehicle_id, sim):
         return sim.vehicle_index.vehicle_by_id(vehicle_id).trip_meter_sensor()
 
-    def step_agent_sensors(self, sim):
-        for agent_id in self.active_agents:
-            for vehicle_id in sim.vehicle_index.vehicle_ids_by_actor_id(
-                agent_id, include_shadowers=True
-            ):
-                sensor_state = sim.vehicle_index.sensor_state_for_vehicle_id(vehicle_id)
-                Sensors.step(self, sensor_state)
+    def step_sensors(self, sim):
+        for vehicle_id, sensor_state in sim.vehicle_index.sensor_states().items():
```

Comment:
Can you use `agent_vehicle_ids` instead that way we avoid returning the entire sensor state dictionary?

Reply:
agent_vehicle_ids do not include shadowed agents, which was the original problem that causes https://github.com/huawei-noah/SMARTS/issues/99